### PR TITLE
MAINT: Implement option to oneDAL's lib path to sklearnex's RPATH

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR adds the path to the .so files of oneDAL to the RPATH of the generated .so files for `daal4py` and `onedal` in sklearnex, so that env. variables pointing to oneDAL would not need to be defined before the library can be loaded.

Note: this is an experimental PR, I'm not yet sure of the full implications.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

**Performance**

Not applicable.

### CODE REVIEW

File: scripts/CMakeLists.txt [hunk 1]
```diff
@@ -40,7 +40,7 @@ elseif(UNIX)
         set(CMAKE_INSTALL_RPATH "@loader_path/../../../")
     else()
         set(ONEDAL_PY_LINK_OPTIONS "-Wl,-z,noexecstack,-z,relro,-z,now,-fstack-protector-strong,-fno-strict-overflow,-fno-delete-null-pointer-checks,-fwrapv")
-        set(CMAKE_INSTALL_RPATH "$ORIGIN/../../../")
+        set(CMAKE_INSTALL_RPATH "$ORIGIN/../../../:${oneDAL_LIBRARY_DIR}")
```
Comment (Alexsandruss) [inline scripts/CMakeLists.txt:43]: Usage of absolute paths has two issues if package is built for public distribution:
- It is non-portable
- It is absolute path on building machine which can be a security issue

Comment (david-cortes-intel) [reply scripts/CMakeLists.txt:43]: Thanks for looking into it.

What would you think of enabling it for non-release builds through an optional argument?

I think it shouldn't be much of an issue that it isn't portable, because it would still be able to load the library if it finds it elsewhere (e.g. in the first RPATH entry which is relative, or in `LD_LIBRARY_PATH`).

Comment (Alexsandruss) [reply scripts/CMakeLists.txt:43]: Optional argument is fine.

Comment (napetrov) [reply scripts/CMakeLists.txt:43]: More over we would need to ensure that we are not adding those accidentally to release builds

Comment (david-cortes-intel) [reply scripts/CMakeLists.txt]: Made it into an optional behavior that can be triggered by passing an argument `--dev` to the `setup.py` script.

I think this should be a sufficiently high barrier to avoid accidents, because it's not a recognized built-in argument for setuptools and thus calls to setup.py with that argument would error out before this PR.

File: setup.py [hunk 2]
```diff
@@ -41,6 +41,13 @@
 from scripts.package_helpers import get_packages_with_tests
 from scripts.version import get_onedal_shared_libs, get_onedal_version
 
+USE_ABS_RPATH = False
+abs_rpath_args = ["abs-rpath", "-abs-rpath", "--abs-rpath"]
```
Comment (Alexsandruss) [inline setup.py:45]: Why three types of prefix? Double is closest to a standard.

Comment (david-cortes-intel) [reply setup.py]: Modified to accept only a double-dashed argument.

**Review by Alexsandruss — APPROVED** [review]

**Comment by icfaust** [comment]
@david-cortes-intel does this add to the daal4py so file? im having trouble seeing it here and it says it does in the description
